scripts/dts/extract/globals.py

```diff
@@ -331,6 +331,44 @@ def build_cell_array(prop_array):
 
     return ret_array
 
+def gpio_map_lookup(cell_parent, gpio_index):
```

Comment:
It looks like the only thing this function uses `cell_parent` for is to look up `reduced[cell_parent]['props']`. Do you think it could just take the properties directly as an argument instead (something like `cell_parent_props` maybe)?